src/plugins/data_source_management/public/components/validation/datasource_form_validation.test.ts

```diff
@@ -49,6 +53,16 @@ describe('DataSourceManagement: Form Validation', () => {
       const result = performDataSourceFormValidation(form, [], '');
       expect(result).toBe(false);
     });
+    test('should fail validation when endpoint path is not empty', () => {
+      form.endpoint = mockDataSourceAttributesWithInvalidEndpoint.endpoint;
+      const result = performDataSourceFormValidation(form, [], '');
+      expect(result).toBe(false);
+    });
+    test('should NOT fail validation when endpoint path is empty', () => {
+      form.endpoint = mockDataSourceAttributesWithValidEndpoint.endpoint;
+      const result = performDataSourceFormValidation(form, [], '');
+      expect(result).toBe(false);
```

Comment:
The expected result here should be true.

Reply:
Good catch~